### PR TITLE
WIP: Experimental atomic CSS approach

### PR DESCRIPTION
**Pull request by jxnblk**

### CODE REVIEW

File: src/hash.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+const hash = (str) => {
+  let val = 5381
+  let i = str.length
+
+  while (i) {
+    val = (val * 33) ^ str.charCodeAt(--i)
```
Comment (asilvas) [inline src/hash.js]: What method of hashing is this?

File: src/index.js [hunk 2]
```diff
@@ -1,62 +1,170 @@
...
-    .forEach(r => {
-      cache.push(r.id)
-      sheet.insert(r.css)
-    })
-
-  return classNames.reduce((a, b) => {
-    if (a.indexOf(b) > -1) return a
+// Insert rule into stylesheet and cache
+const insert = rule => {
+  if (cache.indexOf(rule.className) > -1) {
```
Comment (asilvas) [inline src/index.js]: Why use array instead of hash? This seems like it'd be very wasteful, especially for larger applications with hundreds of classes.

Comment (jxnblk) [inline src/index.js]: Not sure I understand... When you say *hash* do you mean use an object instead? Would that be faster than an array?

**Comment by jxnblk** [comment]
Will probably close this in favor of #34

**Pull request closed without merging**